pkg/actuators/machine/reconciler.go

```diff
@@ -187,11 +188,48 @@ func (r *Reconciler) update() error {
 
 	klog.Infof("Updated machine %s", r.machine.Name)
 
-	r.machineScope.setProviderStatus(newestInstance, conditionSuccess())
+	domainNames, err := r.getCustomDomainFromDHCP(newestInstance.VpcId)
+
+	if err != nil {
+		return err
+	}
+
+	r.machineScope.setProviderStatus(newestInstance, conditionSuccess(), domainNames)
 
 	return r.requeueIfInstancePending(newestInstance)
 }
 
+func (r *Reconciler) getCustomDomainFromDHCP(vpcID *string) ([]string, error) {
```

Comment:
can we at minimum have a unit for this function? e.g similar to `TestGetBlockDeviceMappings`